core/backend/src/ChangesetECAdaptor.ts

```diff
@@ -0,0 +1,778 @@
+                      (
+                        SELECT
+                          JSON_OBJECT(
+                            'table', tb.Name,
+                            'column', cc.Name,
+                            'type',
+                            (
+                              CASE cc.Type
+                                WHEN 0 THEN 'Any'
+                                WHEN 1 THEN 'Boolean'
+                                WHEN 2 THEN 'Blob'
+                                WHEN 3 THEN 'Timestamp'
+                                WHEN 4 THEN 'Real'
+                                WHEN 5 THEN 'Integer'
+                                WHEN 6 THEN 'Text'
+                              END
+                            ),
+                            'columnKind',
+                            (
+                              CASE cc.ColumnKind
+                                WHEN 0 THEN 'Default'
+                                WHEN 1 THEN 'Id'
+                                WHEN 2 THEN 'ClassId'
+                                WHEN 4 THEN 'SharedData'
+                              END
+                            ),
+                            'accessString', pp0.AccessString,
+                            'isVirtual', cc.IsVirtual OR tb.Type = 4
+                          ) columnJson
+                        FROM [ec_PropertyMap] [pm0]
+                          JOIN [ec_Column] [cc] ON [cc].[Id] = [pm0].[ColumnId]
+                          JOIN [ec_Table] [tb] ON [tb].[Id] = [cc].[TableId]
+                          JOIN [ec_PropertyPath] [pp0] ON [pp0].[Id] = [pm0].[PropertyPathId]
+                        WHERE
+                          [pp0].[RootPropertyId] = pt.Id AND pm0.ClassId = cs.Id
+                      )
+                  )
+                ) propJson
+              FROM [ec_PropertyMap] [pm]
+                JOIN [ec_PropertyPath] [pp] ON [pp].[Id] = [pm].[PropertyPathId]
+                JOIN [ec_Property] [pt] ON [pt].[Id] = [pp].[RootPropertyId]
+              WHERE
+                pm.ClassId = cs.Id
+              GROUP BY
+                pt.Id
+            )
+        )
+      ) classDef
+    FROM [ec_Class] [cs]
+      JOIN [ec_ClassMap] [cm] ON [cm].[ClassId] = [cs].[Id]
+      JOIN [ec_Schema] [ss] ON [ss].[Id] = [cs].[SchemaId]
+    WHERE
+      [cs].[Id] = ?
+    `;
+
+    return this.db.withPreparedSqliteStatement(sql, (stmt) => {
+      stmt.bindId(1, classId);
+      if (stmt.step() === DbResult.BE_SQLITE_ROW) {
+        const classMap = JSON.parse(stmt.getValueString(0), (key, value) => {
+          if (value === null) {
+            return undefined;
+          }
+          if (key === "isVirtual") {
+            return value === 0 ? false : true;
+          }
+          return value;
+        }) as IClassMap;
+
+        this._cachedClassMaps.set(classId, classMap);
+        return classMap;
+      }
+      return undefined;
+    });
+  }
+}
+
+/** @beta */
+export interface MetaData {
```

Comment:
Need a more specific name than "MetaData".

Reply:
I did not intent to export these classes. They are internal i will remove export from it.

Reply:
Ah wait this is something i need to export I will rename it.

Reply:
renamed